src/Step.cc

```diff
@@ -109,21 +109,24 @@ void Step::copy_used_variables_from_lua_to_context(const sol::state& lua, Contex
     }
 }
 
-bool Step::execute_impl(Context& context, CommChannel* comm, StepIndex index)
+bool Step::execute_impl(Context& context, CommChannel* comm, StepIndex index,
+    std::string step_setup)
 {
     sol::state lua;
 
     open_safe_library_subset(lua);
     install_custom_commands(lua, context);
 
-    if (context.lua_init_function)
-        context.lua_init_function(lua);
+    if (context.lua_step_setup)
+        context.lua_step_setup(lua);
 
     install_timeout_and_termination_request_hook(lua, Clock::now(), get_timeout(), index,
                                                  comm);
 
+    auto script = allow_step_setup() ? step_setup + '\n' + get_script() : get_script();
```

Comment:
String concatenation is one way of doing it, and it even preserves *local* variables.
We could also call `execute_lua_script()` twice with the same state; in this case only the globals would survive. I have a slight preference for the latter, this gives users a chance to hide away some of their variables.

Reply:
Ok. Interesting approach! I must think about that...

Reply:
I think it should be as easy as this:
```cpp
    if (executes_script())
    {
        const auto result_or_error = execute_lua_script(lua, get_script());
        if (std::holds_alternative<std::string>(result_or_error))
            throw Error("Error in Lua setup script: " + std::get<std::string>(result_or_error));
    }

    copy_used_variables_from_context_to_lua(context, lua);
    const auto result_or_error = execute_lua_script(lua, script);
    copy_used_variables_from_lua_to_context(lua, context);
```

We can discuss how we want to handle the context variables, but I think it is OK not to export them if already the setup script fails.

Reply:
Oh oh I have the feeling that I don't see the wood for the trees. You are right. Using the identical Lua state after the setup script already has the previous variables and functions set. My mistake. Now the tests runs like a charm.

BTW I reduced the longish exception message to 

```c++
throw Error(gul14::cat("[setup] ", std::get<std::string>(result_or_error)));
```